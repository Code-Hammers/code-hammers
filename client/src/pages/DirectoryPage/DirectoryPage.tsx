import React, { useEffect, useState } from "react";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { fetchAlumni } from "../../features/alumni/alumniSlice";

const DirectoryPage = (): JSX.Element => {
  const dispatch = useAppDispatch();
  const { alumni, status, page, totalPages } = useAppSelector(
    (state) => state.alumni
  );
  const [nameSearch, setNameSearch] = useState("");
  const [companySearch, setCompanySearch] = useState("");

  useEffect(() => {
    dispatch(fetchAlumni({ page, name: nameSearch, company: companySearch }));
  }, [dispatch, page, nameSearch, companySearch]);

  const handlePreviousPage = () => {
    if (page > 1) {
      dispatch(
        fetchAlumni({
          page: page - 1,
          name: nameSearch,
          company: companySearch,
        })
      );
    }
  };

  const handleNextPage = () => {
    if (page < totalPages) {
      dispatch(
        fetchAlumni({
          page: page + 1,
          name: nameSearch,
          company: companySearch,
        })
      );
    }
  };

  return (
    <div className="min-h-screen bg-gray-900 text-white flex flex-col items-center justify-start pt-20 p-4">
      <div className="sticky mt-20 z-10 bg-gray-900 w-full text-center py-4">
        <h1 className="text-4xl font-extrabold mb-4">Alumni Directory</h1>
        <div className="flex justify-center space-x-4 mb-4">
          <input
            type="text"
            value={nameSearch}
            onChange={(e) => setNameSearch(e.target.value)}
            placeholder="Search by name"
            className="mb-4 p-2 text-gray-900"
          />
          <input
            type="text"
            value={companySearch}
            onChange={(e) => setCompanySearch(e.target.value)}
            placeholder="Search by company"
            className="mb-4 p-2 text-gray-900"
          />
        </div>
      </div>
      <div className="w-full max-w-3xl">
        <div className="grid grid-cols-3 text-center mb-2">
          <strong>Name</strong>
          <strong>Company</strong>
          <strong>Email</strong>
        </div>
        <div className="overflow-auto" style={{ maxHeight: "60vh" }}>
          {status === "loading" ? (
            <p>Loading...</p>
          ) : (
            alumni.map((alum) => (
              <div key={alum._id} className="grid grid-cols-3 text-center py-2">
                <div>{alum.name}</div>
                <div>{alum.company}</div>
                <div>{alum.email}</div>
              </div>
            ))
          )}
        </div>
        <div className="py-4 flex justify-between">
          <button onClick={handlePreviousPage} disabled={page <= 1}>
            Previous
          </button>
          <button onClick={handleNextPage} disabled={page >= totalPages}>
            Next
          </button>
        </div>
      </div>
    </div>
  );
};

export default DirectoryPage;
