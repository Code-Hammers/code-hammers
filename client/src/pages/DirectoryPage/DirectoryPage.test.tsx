import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import DirectoryPage from './DirectoryPage';

interface State {
  alumni: {
    alumni: any[];
    status: 'idle' | 'loading' | 'failed';
    page: number;
    totalPages: number;
  };
}

const mockStore = configureStore<State>([]);
const initialState: State = {
  alumni: {
    alumni: [],
    status: 'idle',
    page: 1,
    totalPages: 1,
  },
};

describe('MainPage Component', () => {
  it('renders correctly', () => {
    const store = mockStore(initialState);
    const tree = create(
      <Provider store={store}>
        <DirectoryPage />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
