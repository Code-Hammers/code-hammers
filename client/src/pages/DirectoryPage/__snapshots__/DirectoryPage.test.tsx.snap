// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Component renders correctly 1`] = `
<div
  className="min-h-screen bg-gray-100 flex flex-col items-center justify-center"
>
  <h1
    className="text-4xl font-extrabold mb-4"
  >
    Directory Page
  </h1>
  <h1>
    Welcome 
     !
  </h1>
</div>
`;
